File: src/main/java/top/zbeboy/isy/service/system/AuthoritiesServiceImpl.java
package top.zbeboy.isy.service.system;


import lombok.extern.slf4j.Slf4j;
import org.jooq.DSLContext;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.RememberMeAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.ObjectUtils;
import top.zbeboy.isy.domain.tables.pojos.Authorities;
import top.zbeboy.isy.domain.tables.records.AuthoritiesRecord;

import java.util.List;

import static top.zbeboy.isy.domain.Tables.AUTHORITIES;

/**
 * Created by lenovo on 2016-02-21.
 */
@Slf4j
@Service("authoritiesService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
public class AuthoritiesServiceImpl implements AuthoritiesService {

    private final DSLContext create;

    @Autowired
    public AuthoritiesServiceImpl(DSLContext dslContext) {
        this.create = dslContext;
    }

    @Override
    public List<AuthoritiesRecord> findByUsername(String username) {
        return create.selectFrom(AUTHORITIES).where(AUTHORITIES.USERNAME.eq(username)).fetch();
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    @Override
    public void save(Authorities authorities) {
        create.insertInto(AUTHORITIES)
                .set(AUTHORITIES.USERNAME, authorities.getUsername())
                .set(AUTHORITIES.AUTHORITY, authorities.getAuthority())
                .execute();
    }

    @Override
    public void deleteByUsername(String username) {
        create.deleteFrom(AUTHORITIES)
                .where(AUTHORITIES.USERNAME.eq(username))
                .execute();
    }

    @Override
    public void deleteByAuthorities(String authorities) {
        create.deleteFrom(AUTHORITIES)
                .where(AUTHORITIES.AUTHORITY.eq(authorities))
                .execute();
    }

    @Override
    public boolean isRememberMeAuthenticated() {
        Authentication authentication =
                SecurityContextHolder.getContext().getAuthentication();
        return !ObjectUtils.isEmpty(authentication) && RememberMeAuthenticationToken.class.isAssignableFrom(authentication.getClass());
    }
}

File: src/main/java/top/zbeboy/isy/service/system/AuthoritiesServiceImpl.kt
package top.zbeboy.isy.service.system

import org.jooq.DSLContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.RememberMeAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import org.springframework.util.ObjectUtils
import top.zbeboy.isy.domain.tables.pojos.Authorities
import top.zbeboy.isy.domain.tables.records.AuthoritiesRecord

import top.zbeboy.isy.domain.Tables.AUTHORITIES
/**
 * Created by zbeboy 2017-11-17 .
 **/
@Service("authoritiesService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
open class AuthoritiesServiceImpl @Autowired constructor(dslContext: DSLContext) : AuthoritiesService{

    private val create: DSLContext = dslContext

    override fun findByUsername(username: String): List<AuthoritiesRecord> {
        return create.selectFrom<AuthoritiesRecord>(AUTHORITIES).where(AUTHORITIES.USERNAME.eq(username)).fetch()
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    override fun save(authorities: Authorities) {
        create.insertInto<AuthoritiesRecord>(AUTHORITIES)
                .set(AUTHORITIES.USERNAME, authorities.username)
                .set(AUTHORITIES.AUTHORITY, authorities.authority)
                .execute()
    }

    override fun deleteByUsername(username: String) {
        create.deleteFrom<AuthoritiesRecord>(AUTHORITIES)
                .where(AUTHORITIES.USERNAME.eq(username))
                .execute()
    }

    override fun deleteByAuthorities(authorities: String) {
        create.deleteFrom<AuthoritiesRecord>(AUTHORITIES)
                .where(AUTHORITIES.AUTHORITY.eq(authorities))
                .execute()
    }

    override fun isRememberMeAuthenticated(): Boolean {
        val authentication = SecurityContextHolder.getContext().authentication
        return !ObjectUtils.isEmpty(authentication) && RememberMeAuthenticationToken::class.java.isAssignableFrom(authentication.javaClass)
    }

}